packages/flutter/lib/src/material/text_field.dart

```diff
@@ -195,6 +245,15 @@ class TextField extends StatefulWidget {
     description.add(new DiagnosticsProperty<bool>('obscureText', obscureText, defaultValue: false));
     description.add(new DiagnosticsProperty<bool>('autocorrect', autocorrect, defaultValue: false));
     description.add(new IntProperty('maxLines', maxLines, defaultValue: 1));
+    description.add(new IntProperty('maxLength', maxLength, defaultValue: null));
+    description.add(
+      new FlagProperty(
+        'maxLengthEnforced',
+        value: maxLengthEnforced,
+        ifTrue: 'enforced',
+        ifFalse: 'not enforced',
+      ),
+    );
```

Comment:
this should be all on one line for clarity. Also you probably only want the ifTrue, and you want it to make sense alone ("max length enforced"), since the name isn't shown for FlapPropertys..

Reply:
FWIW, Jacob and I ended up deciding that it was just better overall if all the diagnostics were always on one line. We tried it various ways on his giant PR that added the majority of them and it really ended up being the better solution.